### ISSUE TITLE
Library dll files not found

### ISSUE THREAD

**Issue by gunitinug** [Winetricks/winetricks#1806]
Hello. I got the latest version of winetricks. When I do `winetricks vcrun6sp6` I get errors saying various dll's are missing: for example, `024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\ADVAPI32.dll") not found`. But if I do `locate msvcrt.dll` I get:
```
/home/logan/.PlayOnLinux/wine/linux-x86/5.22/lib/wine/msvcrt.dll
/home/logan/.PlayOnLinux/wine/linux-x86/6.15/lib/wine/i386-windows/msvcrt.dll
/home/logan/.PlayOnLinux/wine/linux-x86/6.15-staging/lib/wine/i386-windows/msvcrt.dll
/home/logan/.PlayOnLinux/wineprefix/my_prefix/drive_c/windows/system32/msvcrt.dll
/usr/lib/i386-linux-gnu/wine/msvcrt.dll.so
/usr/lib/i386-linux-gnu/wine/fakedlls/msvcrt.dll
```
So why is winetricks not finding all the dll files that I already have in c:/windows/system32 in my wine prefix?

```
Executing mkdir -p /home/logan/.PlayOnLinux//wineprefix
Using winetricks 20210206-next - sha256sum: 0769bbcdbd92c407f4eacaa85acc1339f607dbeafe2febd1be0912034c7af3a1 with wine-6.15 and WINEARCH=win32
Executing w_do_call vcrun6sp6
Executing mkdir -p /home/logan/.PlayOnLinux//wineprefix
Executing load_vcrun6sp6 
Executing cd /home/logan/.cache/winetricks/vcrun6sp6
Downloading https://www.ddsystem.com.br/update/setup/vb6+sp6/VS6SP6.EXE to /home/logan/.cache/winetricks/vcrun6sp6
--2021-08-19 16:13:27--  https://www.ddsystem.com.br/update/setup/vb6+sp6/VS6SP6.EXE
Resolving www.ddsystem.com.br (www.ddsystem.com.br)... 192.185.213.53
Connecting to www.ddsystem.com.br (www.ddsystem.com.br)|192.185.213.53|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 63799912 (61M) [application/x-msdownload]
Saving to: 'VS6SP6.EXE’

     0K .......... .......... .......... .......... ..........  0%  122K 8m28s
    50K .......... .......... .......... .......... ..........  0%  244K 6m21s

skipped...

   100K .......... 62200K .......... .......... .......... .......... .......... 99% 10.0M 0s
 62250K .......... .......... .......... .......... .......... 99% 24.6M 0s
 62300K ....                                                  100% 20.4M=7.6s

2021-08-19 16:13:35 (8.05 MB/s) - 'VS6SP6.EXE’ saved [63799912/63799912]

Executing cd /home/logan/.PlayOnLinux/wineprefix/kuf/drive_c
Executing cabextract -q /home/logan/.cache/winetricks/vcrun6sp6/VS6SP6.EXE -d /home/logan/.PlayOnLinux//wineprefix/kuf/dosdevices/c:/windows/temp -F vcredist.exe
Executing cd /home/logan/.PlayOnLinux//wineprefix/kuf/dosdevices/c:/windows/temp
Executing rm -f /home/logan/.PlayOnLinux//wineprefix/kuf/dosdevices/c:/windows/system32/comcat.dll
Executing rm -f /home/logan/.PlayOnLinux//wineprefix/kuf/dosdevices/c:/windows/system32/msvcrt.dll
Executing rm -f /home/logan/.PlayOnLinux//wineprefix/kuf/dosdevices/c:/windows/system32/oleaut32.dll
Executing rm -f /home/logan/.PlayOnLinux//wineprefix/kuf/dosdevices/c:/windows/system32/olepro32.dll
Executing rm -f /home/logan/.PlayOnLinux//wineprefix/kuf/dosdevices/c:/windows/system32/stdole2.tlb
002c:err:wineboot:process_run_key Error running cmd L"C:\\windows\\system32\\winemenubuilder.exe -a -r" (126).
006c:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
006c:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
006c:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
006c:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
006c:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
006c:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0084:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
0084:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
0084:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
0084:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
0084:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0084:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
002c:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
002c:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
002c:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
002c:fixme:font:get_name_record_codepage encoding 29 not handled, platform 1.
002c:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
002c:fixme:font:get_name_record_codepage encoding 20 not handled, platform 1.
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\ADVAPI32.dll") not found
0024:err:module:import_dll Library ADVAPI32.dll (which is needed by L"C:\\windows\\temp\\vcredist.exe") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\GDI32.dll") not found
0024:err:module:import_dll Library GDI32.dll (which is needed by L"C:\\windows\\temp\\vcredist.exe") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\USER32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\gdi32.dll") not found
0024:err:module:import_dll Library gdi32.dll (which is needed by L"C:\\windows\\system32\\USER32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\setupapi.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\rpcrt4.dll") not found
0024:err:module:import_dll Library rpcrt4.dll (which is needed by L"C:\\windows\\system32\\setupapi.dll") not found
0024:err:module:import_dll Library setupapi.dll (which is needed by L"C:\\windows\\system32\\USER32.dll") not found
0024:err:module:import_dll Library USER32.dll (which is needed by L"C:\\windows\\temp\\vcredist.exe") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\COMCTL32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\gdi32.dll") not found
0024:err:module:import_dll Library gdi32.dll (which is needed by L"C:\\windows\\system32\\COMCTL32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\imm32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\user32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\gdi32.dll") not found
0024:err:module:import_dll Library gdi32.dll (which is needed by L"C:\\windows\\system32\\user32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\setupapi.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\rpcrt4.dll") not found
0024:err:module:import_dll Library rpcrt4.dll (which is needed by L"C:\\windows\\system32\\setupapi.dll") not found
0024:err:module:import_dll Library setupapi.dll (which is needed by L"C:\\windows\\system32\\user32.dll") not found
0024:err:module:import_dll Library user32.dll (which is needed by L"C:\\windows\\system32\\imm32.dll") not found
0024:err:module:import_dll Library imm32.dll (which is needed by L"C:\\windows\\system32\\COMCTL32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\user32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\gdi32.dll") not found
0024:err:module:import_dll Library gdi32.dll (which is needed by L"C:\\windows\\system32\\user32.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\setupapi.dll") not found
0024:err:module:import_dll Library msvcrt.dll (which is needed by L"C:\\windows\\system32\\advapi32.dll") not found
0024:err:module:import_dll Library advapi32.dll (which is needed by L"C:\\windows\\system32\\rpcrt4.dll") not found
0024:err:module:import_dll Library rpcrt4.dll (which is needed by L"C:\\windows\\system32\\setupapi.dll") not found
0024:err:module:import_dll Library setupapi.dll (which is needed by L"C:\\windows\\system32\\user32.dll") not found
0024:err:module:import_dll Library user32.dll (which is needed by L"C:\\windows\\system32\\COMCTL32.dll") not found
0024:err:module:import_dll Library COMCTL32.dll (which is needed by L"C:\\windows\\temp\\vcredist.exe") not found
0024:err:module:LdrInitializeThunk Importing dlls for L"C:\\windows\\temp\\vcredist.exe" failed, status c0000135
------------------------------------------------------
warning: vcrun6sp6 installation failed
------------------------------------------------------
```

**Comment by austin987** [comment]
Thanks @gverm, looks like wmp/windowscodecs are also affected. vb6run works fine here (in 6.15), but yeah, may break in the future/easy enough to fix.

I've pushed fixes for vb6run/vcrun6/vcrun6sp6 to https://github.com/austin987/winetricks/tree/issue-1806-vcrun6sp6; ie6/ie7/wmp9/wmp10 I haven't looked at yet, likely tricker. Based on https://bugs.winehq.org/show_bug.cgi?id=21009#c3, I suspect some of them may not be fixable.

windowscodecs from a quick look does more than simply extract dlls, may or may not be doable manually, still looking.

**Comment by gverm** [comment]
@austin987 Wouldn't it be better to just overwrite the builtins and still let the installer run? That seems like it might be good for the others as well.

If that's not an option, we could go with the WINEBOOTSTRAPMODE method I mentioned in #1756, since we know these installers run without these dlls present.

**Comment by austin987** [comment]
@gverm possibly, yes, I haven't tried yet :)

Also, thanks for the reminder about WINEBOOTSTRAPMODE :+1:

**Comment by prez** [comment]
I can reproduce this. See https://github.com/Winetricks/winetricks/issues/1810.
ie8 seems broken too.

**Comment by gverm** [comment]
@austin987 I opened #1813 to go with option 1 (only partially manually extract the dlls).
I checked out windowscodecs as well, but that one's trickier, as the problem there is that the installer won't run correctly if the files exists. I guess a full manual approach is needed there.

I would avoid WINEBOOTSTRAPMODE as much as possible, since I recall Alexandre mentioning that it won't be around forever.
